acados/sim/sim_erk_integrator.c

```diff
@@ -434,12 +434,42 @@ void sim_erk_opts_update(void *config_, void *dims, void *opts_)
  * memory
  ************************************************/
 
-int sim_erk_memory_calculate_size(void *config, void *dims, void *opts_) { return 0; }
+int sim_erk_memory_calculate_size(void *config, void *dims, void *opts_)
+{
+    return 0;
+}
+
+
 void *sim_erk_memory_assign(void *config, void *dims, void *opts_, void *raw_memory)
 {
     return NULL;
 }
 
+int sim_erk_memory_set(void *config_, void *dims_, void *mem_, const char *field, void *value)
+{
+    // sim_config *config = config_;
+    int status = ACADOS_FAILURE;
+
+    return status;
+}
+
+int sim_erk_memory_set_to_zero(void *config_, void * dims_, void *opts_, void *mem_, const char *field)
+{
+    int status = ACADOS_SUCCESS;
+
+    if (!strcmp(field, "guesses"))
+    {
+        // no guesses/initialization in ERK
+    }
+    else
+    {
+        status = ACADOS_FAILURE;
```

Comment:
Error handling is not handled systematically in acados, and just returning a chain of `ACADOS_FAILURE` would make very difficult to debug something like a wrong field.

Therefore, the way to go is rather to `printf("\nerror: sim_erk_memory_set_to_zero: wrong field %s\n", field);`, which tells exactly what and where is the issue.
Eventually all setters/getters should do so, but for now let's at least stick to this way for the new ones.

Reply:
done, also for other setters & checks